charts/opentelemetry-collector/templates/NOTES.txt

```diff
@@ -40,3 +40,9 @@
 {{- if .Values.presets.kubernetesEvents.enabled }}
 [WARNING] The 'k8sobjects' is a ALPHA receiver and may be changed anytime.
 {{ end }}
+
+{{- if .Values.service.serviceForDaemonset }}
+{{- if not (eq .Values.mode "daemonset") }}
+{{- fail "Enabling serviceForDaemonset is only suitable for daemonset mode. Please use daemonset mode to enable it."}}
+{{ end }}
+{{ end }}
```

Comment:
Please also add a warning is mode is daemonset and the mode is set to cluster.

Reply:
Done! :)